Allow mocha to be installed globally or on any parent folder node_mod…

andreesteve opened the pull request:
#### Issue
#403  (Allow Mocha to be installed Globally) 

##### Fix
This PR changes how mocha module is found. Instead of only looking on the project folder node_modules, I transverse the file system tree looking at node_modules folder on parent directories. If the module is not found in any parent node_modules folder, then a global require is issued.

##### Testing
1. No globally installed mocha package, no mode_modules on project folder or parent folder -> Expects error message (module not found).
2. Mocha locally installed (previous behavior) -> Expects to loads local module as before.
3. Mocha installed on some parent node_modules folder -> Expects to load module from parent node_module folder.
4. Mocha not installed locally or on parent folder, but installed globally -> Expects to load global module. (**NOTE:**: on Windows one might need to set the NODE_PATH environment variable to point to NPM global install folder).

msftclas commented:
Hi __@andreesteve__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.microsoft.com.
    </span>

TTYL, MSBOT;

msftgits commented:
Hi, I am closing and re-opening this PR to bump the CLA bot. Sorry for the inconvenience!

msftclas commented:
Hi __@andreesteve__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, MSBOT;

paulvanbrenk commented:
@andreesteve  looks like this fell through the cracks. Do you mind singing the CLA so I can merge this in. 

One change I would like to see is to add the error message back in case the script can't find the mocha install.

andreesteve commented:
@paulvanbrenk , I've rebased the change on top of current master (the file was moved).

The error message, in case mocha is not found, is still there, on line 273. I think I removed the EOL not sure why - this was a while ago - so the formatting looks a bit different, but the error is still reported for such case.

It shows the CLA signed here, not sure how I sign it again. Do you have the link? I can try it again.

paulvanbrenk commented:
Thanks, I see it now.. looks good to me. May take me another week to actually get this in, but it will go in.

Thanks for the contribution!

marklam commented:
I've just hit this limitation, it would me nice if this PR was merged.

joj commented:
Overridden by more recent changes. Closing.
